redfish-core/lib/processor.hpp

```diff
@@ -975,6 +995,280 @@ inline void
+        std::bind_front(afterGetCpuCoreDataByService, asyncResp, objPath));
+}
+
+inline void aferGetSubProcessorData(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& processorId, const std::string& coreId,
+    const boost::system::error_code& ec,
+    const dbus::utility::MapperGetSubTreeResponse& subtree)
+{
+    if (ec)
+    {
+        BMCWEB_LOG_DEBUG("DBUS response error, ec: {}", ec.value());
+        // No processor objects found by mapper
+        if (ec.value() == boost::system::errc::io_error)
+        {
+            messages::resourceNotFound(
+                asyncResp->res, "#Processor.v1_18_0.Processor", processorId);
+            return;
+        }
+
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    for (const auto& object : subtree)
+    {
+        if (sdbusplus::message::object_path(object.first).filename() != coreId)
+        {
+            continue;
+        }
+
+        asyncResp->res.jsonValue["@odata.type"] =
+            "#Processor.v1_18_0.Processor";
+        asyncResp->res.jsonValue["@odata.id"] = boost::urls::format(
+            "/redfish/v1/Systems/system/Processors/{}/SubProcessors/{}",
+            processorId, coreId);
+        asyncResp->res.jsonValue["Name"] = "SubProcessor";
+        asyncResp->res.jsonValue["Id"] = coreId;
+
+        for (const auto& service : object.second)
+        {
+            getCpuCoreDataByService(asyncResp, service.first, object.first);
+            break;
+        }
+        return;
+    }
+
+    // Object not found
+    messages::resourceNotFound(asyncResp->res, "#Processor.v1_18_0.Processor",
+                               coreId);
+}
+
+inline void
+    getSubProcessorData(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                        const std::string& processorId,
+                        const std::string& coreId)
+{
+    auto callback = [asyncResp, processorId,
+                     coreId](const std::string& cpuPath) {
+        dbus::utility::getSubTree(cpuPath, 0, cpuCoreIfaces,
+                                  std::bind_front(aferGetSubProcessorData,
+                                                  asyncResp, processorId,
+                                                  coreId));
+    };
+
+    getProcessorPaths(asyncResp, processorId, std::move(callback));
+}
+
+inline void afterGetSubProcessorMembers(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& processorId, const boost::system::error_code& ec,
+    const dbus::utility::MapperGetSubTreePathsResponse& subTreePaths)
+{
+    if (ec)
+    {
+        BMCWEB_LOG_ERROR("DBUS response error: {}", ec.value());
+        // No processor objects found by mapper
+        if (ec.value() == boost::system::errc::io_error)
+        {
+            messages::resourceNotFound(
+                asyncResp->res, "#Processor.v1_18_0.Processor", processorId);
+            return;
+        }
+
```

Comment:
I think L1225 is better to be moved here

Reply:
+1